internal/pkg/cli/task_run.go

```diff
@@ -134,6 +163,117 @@ func (o *runTaskOpts) Ask() error {
 	return nil
 }
 
+// Execute create or update task resources and run the task.
+func (o *runTaskOpts) Execute() error {
+	if o.image == "" && o.dockerfilePath == "" {
+		o.dockerfilePath = "./Dockerfile"
+	}
+
+	if o.imageTag == "" {
+		o.imageTag = "latest"
+	}
+
+	if err := o.getNetworkConfig(); err != nil {
+		return err
+	}
+
+	if err := o.deployTaskResource(); err != nil {
+		return err
+	}
+
+	//if image is not provided, then we build the image and push to ECR repo
+	if o.image == "" {
+		uri, err := o.pushToECRRepo()
+		if err != nil {
+			return err
+		}
+		o.image = fmt.Sprintf(fmtImageURL, uri, o.imageTag)
+
+		// update image to stack
+		if err := o.deployTaskResource(); err != nil {
+			return err
+		}
+	}
+
+	// TODO: kick off task
+	return nil
+}
+
+func (o *runTaskOpts) getNetworkConfig() error {
+	if o.env != config.EnvNameNone {
+		subnets, err := o.ec2.GetSubnetIDsFromAppEnv(o.AppName(), o.env)
+		if err != nil {
+			return fmt.Errorf("get subnet ids: %w", err)
+		}
+
+		securityGroups, err := o.ec2.GetSecurityGroupsFromAppEnv(o.AppName(), o.env)
+		if err != nil {
+			return fmt.Errorf("get security groups: %w", err)
+		}
+
+		o.subnets = subnets
+		o.securityGroups = securityGroups
+
+		return nil
+	}
+
+	// get default subnet ids if not provided
+	if o.subnets == nil {
+		subnetIDs, err := o.ec2.GetDefaultSubnetIDs()
+		if err != nil {
+			return fmt.Errorf("get subnet ids: %w", err)
+		}
+		o.subnets = subnetIDs
+	}
+	return nil
+}
+
+func (o *runTaskOpts) deployTaskResource() error {
+	if err := o.cfn.DeployTask(&deploy.CreateTaskResourcesInput{
+		Name:     o.groupName,
+		Cpu:      o.cpu,
+		Memory:   o.memory,
+		Image:    o.image,
+		TaskRole: o.taskRole,
+		Command:  o.command,
+	}); err != nil {
+		var errChangeSetEmpty *awscfn.ErrChangeSetEmpty
+		if errors.As(err, &errChangeSetEmpty) {
+			return nil
+		}
+		log.Errorf("failed to deploy resources for task: %w", err)
+		return fmt.Errorf("deploy task: %w", err)
+	}
+	return nil
+}
+
+func (o *runTaskOpts) pushToECRRepo() (string, error) {
+	repoName := fmt.Sprintf("%s-%s", "copilot", o.groupName)
+
+	uri, err := o.ecr.GetRepository(repoName)
+	if err != nil {
+		return "", fmt.Errorf("get ECR repository URI: %w", err)
+	}
+
+	if err := o.docker.Build(uri, o.imageTag, o.dockerfilePath); err != nil {
+		return "", fmt.Errorf("build Dockerfile at %s with tag %s: %w", o.dockerfilePath, "", err)
+	}
+
+	auth, err := o.ecr.GetECRAuth()
+	if err != nil {
+		return "", fmt.Errorf("get ECR auth data: %w", err)
+	}
+
+	if err := o.docker.Login(uri, auth.Username, auth.Password); err != nil {
+		return "", fmt.Errorf("login to repo: %w", err)
```

Comment:
```suggestion
		return "", fmt.Errorf("login to repo %s: %w", repoName,  err)
```
maybe this would be more helpful to users?